assets/scripts/build_gallery.py

```diff
@@ -1,11 +1,95 @@
+import ast
 import importlib
 import json
 import logging
 import os
 import sys
+import traceback
+from subprocess import CalledProcessError, CompletedProcess, run
 
-logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
+chandler = logging.StreamHandler(stream=sys.stdout)
+chandler.setLevel(logging.DEBUG)
+chandler.setFormatter(
+    logging.Formatter("%(asctime)s - %(levelname)s - %(message)s", "%Y-%m-%dT%H:%M:%S")
+)
+logger.addHandler(chandler)
+logger.setLevel(logging.DEBUG)
+
+
+def execute_shell_command(command: str) -> int:
+    """
+    Execute a shell (bash in the present case) command from inside Python program.
+
+    While developed independently, this function is very similar to the one, offered in this StackOverflow article:
+    https://stackoverflow.com/questions/30993411/environment-variables-using-subprocess-check-output-python
+
+    :param command: bash command -- as if typed in a shell/Terminal window
+    :return: status code -- 0 if successful; all other values (1 is the most common) indicate an error
+    """
+    cwd: str = os.getcwd()
+
+    path_env_var: str = os.pathsep.join([os.environ.get("PATH", os.defpath), cwd])
+    env: dict = dict(os.environ, PATH=path_env_var)
+
+    status_code: int = 0
+    try:
+        res: CompletedProcess = run(
+            args=["bash", "-c", command],
+            stdin=None,
+            input=None,
+            stdout=None,
+            stderr=None,
+            capture_output=True,
+            shell=False,
+            cwd=cwd,
+            timeout=None,
+            check=True,
+            encoding=None,
+            errors=None,
+            text=None,
+            env=env,
+            universal_newlines=True,
+        )
+        sh_out: str = res.stdout.strip()
+        logger.info(sh_out)
+    except CalledProcessError as cpe:
+        status_code = cpe.returncode
+        sys.stderr.write(cpe.output)
+        sys.stderr.flush()
+        exception_message: str = "A Sub-Process call Exception occurred.\n"
+        exception_traceback: str = traceback.format_exc()
+        exception_message += (
+            f'{type(cpe).__name__}: "{str(cpe)}".  Traceback: "{exception_traceback}".'
+        )
+        logger.error(exception_message)
+
+    return status_code
+
+
+def get_contrib_requirements(filepath):
+    with open(filepath) as file:
+        tree = ast.parse(file.read())
+
+    requirements_info = {"classes": [], "requirements": []}
+    for child in ast.iter_child_nodes(tree):
+        if not isinstance(child, ast.ClassDef):
+            continue
+        requirements_info["classes"] += child.name
```

Comment:
I don't see this being used anywhere?

Reply:
Added a note on the intended future use (and improved its organization). Good call.